### PR TITLE
Implement feature/instrumentation statsbeat

### PR DESCRIPTION
**Pull request by lzchen**
Also add host/endpoint to network statsbeats.

### CODE REVIEW

File: contrib/opencensus-ext-azure/CHANGELOG.md [hunk 1]
```diff
@@ -4,12 +4,14 @@
 
 - Enable AAD authorization via TokenCredential
 ([#1021](https://github.com/census-instrumentation/opencensus-python/pull/1021))
-- Implement attach rate metrics via Statbeat
+- Implement attach rate metrics via Stastbeat
```
Comment (hectorhdzg) [inline contrib/opencensus-ext-azure/CHANGELOG.md:7]:
```suggestion
- Implement attach rate metrics via Statsbeat
```

File: contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/statsbeat_metrics/statsbeat.py [hunk 2]
```diff
@@ -162,8 +185,12 @@ def _get_exception_count_value():
...
 class _StatsbeatMetrics:
 
-    def __init__(self, instrumentation_key):
-        self._instrumentation_key = instrumentation_key
+    def __init__(self, options):
+        self._options = options
+        self._instrumentation_key = options.instrumentation_key
+        self._feature = _StatsbeatFeature.NONE
+        if options.enable_local_storage:
+            self._feature |= _StatsbeatFeature.DISK_RETRY
```
Comment (hectorhdzg) [inline contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/statsbeat_metrics/statsbeat.py:193]: Is this something that can be turned off at some point?, if yes we need to update the value for following Statsbeat

Comment (lzchen) [reply contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/statsbeat_metrics/statsbeat.py:193]: Cannot be turned off without restarting the SDK so feature flag cannot be changed during runtime.

File: contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/statsbeat_metrics/statsbeat.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 import os
 import platform
 import threading
+from opencensus.ext.azure.common import INSTRUMENTATION_KEY
```
[the remark this answers is not in the document]
Comment (lzchen) [reply contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/statsbeat_metrics/statsbeat.py:21]: Will remove.

**Review by hectorhdzg — APPROVED** [review]
LGTM

**Pull request merged**